src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+        Ok(EnumerationArg {
+            name: ArgumentsItem::parse_name(self.name)?,
+            description: ArgumentsItem::parse_description(self.description)?,
+            options: ctx.parse(self.options)?,
+        })
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::EnumerationArg {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::EnumerationArg {
+            name: value.name,
+            description: value.description,
+            options: value.options.into(),
+        }
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::ArgumentsItem::EnumerationArg(value.into())
+    }
+}
+
+impl From<EnumerationArg> for ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
+
+/// List of valid string options
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumOptions(Vec<String>);
+
+impl<C: Context> Parse<C> for simple_extensions::EnumOptions {
+    type Parsed = EnumOptions;
+
+    type Error = EnumOptionsError;
+
+    fn parse(self, _ctx: &mut C) -> Result<EnumOptions, EnumOptionsError> {
+        let options = self.0;
+        if options.is_empty() {
+            return Err(EnumOptionsError::EmptyList);
+        }
+
+        let mut unique_set = std::collections::HashSet::new();
+        for option in options.iter() {
+            if option.is_empty() {
+                return Err(EnumOptionsError::EmptyOption);
+            }
+            if !unique_set.insert(option) {
+                return Err(EnumOptionsError::DuplicatedOption(option.clone()));
+            }
+        }
+
+        Ok(EnumOptions(options))
+    }
+}
+
+impl From<EnumOptions> for simple_extensions::EnumOptions {
+    fn from(value: EnumOptions) -> Self {
+        simple_extensions::EnumOptions(value.0)
+    }
+}
+
+/// Parse errors for [simple_extensions::EnumOptions].
+#[derive(Debug, Error, PartialEq)]
+pub enum EnumOptionsError {
+    /// Empty list.
+    #[error("empty list")]
+    EmptyList,
+
+    /// Duplicated option.
+    #[error("duplicated option: {0}")]
+    DuplicatedOption(String),
+
+    /// Empty option.
+    #[error("empty option")]
+    EmptyOption,
+}
+
+/// Arguments that refer to a data value.
+#[derive(Clone, Debug)]
+pub struct ValueArg {
+    /// A human-readable name for this argument to help clarify use.
+    name: Option<String>,
+
+    /// Additional description for this argument.
+    description: Option<String>,
+
+    /// A fully defined type or a type expression.
+    ///
+    /// todo: implement parsed [simple_extensions::Type].
+    value: simple_extensions::Type,
+
+    /// Whether this argument is required to be a constant for invocation.
+    /// For example, in some system a regular expression pattern would only be accepted as a literal
+    /// and not a column value reference.
+    constant: Option<bool>,
```

Comment:
For this one, do we want the method to return `false` if the option is `None`? Or do  we want to store a `bool` that defaults to `false` if unspecified (what you had before). This is different than the name/description fields in that the spec specifies a default when this field is unset.

Reply:
I think we can return `false` if it is `None`, so that the user does not need to handle the default value separately.

Reply:
I put the description of the default behavior in the comment: [commit](https://github.com/substrait-io/substrait-rs/pull/185/commits/9ed057879faf83f32006224c8314947a9bf4bfed).